### ISSUE TITLE
Error: Refusing to delete - OS X 10.11, npm @3.3.5, node @4.1.0 (works with npm @2.14.6)

### ISSUE THREAD

**Issue by psyrendust** [npm/npm#9818]
## Description

For some reason this seems to work with `npm` `@2.14.6` but fails when using `npm` `@3.3.5`.

## System Info

- OS X 10.11 (fresh install)
- nvm 0.27.1
    - node @4.1.0
    - npm @3.3.5

## Steps to reproduce

- Visit https://github.com/sindresorhus/electron-boilerplate
- Grab the boilerplate using `curl` command:

```
    $ curl -fsSL https://github.com/sindresorhus/electron-boilerplate/archive/master.tar.gz | tar -xz --strip-components 2
```

#### Using npm@3.3.5 (fails)

- Run `npm cache clean && rm -rf ./node_modules` (just to be safe)
- Run `npm install`
- Run `npm run build`
- Error occurs

#### Using npm@2.14.6 (succeeds)

- Run `npm cache clean && rm -rf ./node_modules` (just to be safe)
- Run `npm install`
- Run `npm run build`
- Everything works!

**Comment by psyrendust** [comment]
### npm-debug.log

> TL;DR - Only showing `errors` from npm-debug.log. See below for the full log.

##### Only Errors

```
4006 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/asar is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/asar and not a link
4007 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/electron-download is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/electron-download and not a link
4008 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/detect-indent is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/detect-indent and not a link
4009 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/esparse is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/esprima-fb and not a link
4014 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/handlebars is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/handlebars and not a link
4015 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/indent-string is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/indent-string and not a link
4020 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/json5 is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/json5 and not a link
4021 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/latest-version is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/latest-version and not a link
4022 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/leven is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/leven and not a link
4023 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/ncp is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/ncp and not a link
4024 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/nugget is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/nugget and not a link
4025 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/uuid is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/node-uuid and not a link
4026 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/pretty-bytes is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/pretty-bytes and not a link
4027 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/regenerator is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/regenerator and not a link
4028 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/regjsparser is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/regjsparser and not a link
4029 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/repeating is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/repeating and not a link
4030 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/rimraf is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/rimraf and not a link
4031 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/semver is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/semver and not a link
4032 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/shjs is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/shelljs and not a link
4033 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/strip-json-comments is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/strip-json-comments and not a link
4034 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/user-home is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/user-home and not a link
4035 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/commonize is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/commoner and not a link
4036 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/defs is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/defs and not a link
4037 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/eslint is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/eslint and not a link
4038 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/extract-zip is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/extract-zip and not a link
4039 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/har-validator is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/har-validator and not a link
4040 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/mkdirp is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/mkdirp and not a link
4041 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/js-yaml is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/js-yaml and not a link
4042 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/rc is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/rc and not a link
4043 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/regexpu is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/regexpu and not a link
4044 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/to-double-quotes is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/to-double-quotes and not a link
4045 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/uglifyjs is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/uglify-js and not a link
4046 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/xo is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/xo and not a link
4047 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/to-single-quotes is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/to-single-quotes and not a link
4099 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/jsesc is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/jsesc and not a link
4100 error gentlyRm /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/esvalidate is outside /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/esprima-fb and not a link
4110 error Darwin 15.0.0
4111 error argv "/Users/larrygordon/.nvm/versions/node/v4.1.0/bin/node" "/Users/larrygordon/.nvm/versions/node/v4.1.0/bin/npm" "prune" "--production"
4112 error node v4.1.0
4113 error npm  v3.3.5
4114 error path /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/asar
4115 error code EEXIST
4116 error Refusing to delete: /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/.bin/asar not in /private/var/folders/xc/ywq762sn01d9cmvn79mwqvxw0000gn/T/electron-packager/linux-ia32/App-linux-ia32/resources/app/node_modules/asar
```

**Comment by sindresorhus** [comment]
It seems it's specifically the `--prune` flag in [`electron-messager`](https://github.com/maxogden/electron-packager) which calls `npm prune --production` that is failing in npm@3. Similar to https://github.com/sindresorhus/caprine/issues/19.

**Comment by sindresorhus** [comment]
@vikas-parashar

**Comment by iDuuck** [comment]
Got that error, too! Deleting the `--prune` flag fixes the error.

**Comment by othiym23** [comment]
We will put investigating why `prune --production` crashes on the road map, which means we'll try to get to this in a reasonable time frame.

**Comment by lencioni** [comment]
We are seeing what looks like this issue as well. Here is some of the output:

```
npm ERR! gentlyRm containing path /path/to/project/node_modules/babel isn't under npm's control
npm ERR! gentlyRm containing path /path/to/project/node_modules/json-sass isn't under npm's control
npm ERR! Linux 3.10.0-229.7.2.el7.x86_64
npm ERR! argv "/usr/local/nodejs-5.0.0/bin/node" "/usr/local/bin/npm" "prune" "--production"
npm ERR! node v5.0.0
npm ERR! npm  v3.3.12
npm ERR! path /path/to/project/node_modules/.bin/babel
npm ERR! code EEXIST

npm ERR! Refusing to delete: /path/to/project/node_modules/.bin/babel not in /path/to/project/node_modules/babel
File exists: /path/to/project/node_modules/.bin/babel
Move it away, and try again. 
```

I hope this helps you with this issue.

**Comment by tdooner** [comment]
In case it helps any, in the output @lencioni pasted above, `/path/to/project/node_modules` is actually a symlink.

**Comment by jgrund** [comment]
Seeing this with `npm prune` using `npm@3.4.0` and `node` v5.0.0.

**Comment by zakjan** [comment]
:+1:
